Backend/lib/yudhishthira/src/Lib/Yudhishthira/Types.hs

```diff
@@ -67,7 +73,46 @@ newtype YudhishthiraDecideResp = YudhishthiraDecideResp
 data LogicDomain
   = POOLING
   | FARE_POLICY
-  deriving (Show, Eq, Ord, Read, Generic, ToJSON, FromJSON, ToSchema, ToParamSchema)
+  | DYNAMIC_PRICING DST.ServiceTierType
+  deriving (Eq, Ord, Generic, ToJSON, FromJSON, ToSchema)
+
+generateLogicDomainShowInstances :: [String]
+generateLogicDomainShowInstances =
+  [show POOLING]
+    ++ [show FARE_POLICY]
+    ++ [show (DYNAMIC_PRICING vehicleType) | vehicleType <- vehicleTypes]
+  where
+    vehicleTypes = [COMFY, ECO, PREMIUM, SUV, AUTO_RICKSHAW, HATCHBACK, SEDAN, TAXI, TAXI_PLUS, PREMIUM_SEDAN, BLACK, BLACK_XL, BIKE, AMBULANCE_TAXI, AMBULANCE_TAXI_OXY, AMBULANCE_AC, AMBULANCE_AC_OXY, AMBULANCE_VENTILATOR, SUV_PLUS, DELIVERY_BIKE]
```

Comment:
I don't think we should write it like this, can cause issues if we add more vehicleTypes later.

Reply:
It's just to show it in swagger.

Reply:
No, even db usage show.

Reply:
We should fix it.